src/cryptography/sign.js

```diff
@@ -24,8 +24,16 @@ const signatureHeader = createHeader('SIGNATURE');
 const secondSignatureHeader = createHeader('SECOND SIGNATURE');
 const signatureFooter = createHeader('END LISK SIGNED MESSAGE');
 
-export const signMessageWithPassphrase = (message, passphrase) => {
+const SIGNED_MESSAGE_HEADER = Buffer.from('Lisk Signed Message:\n', 'utf8');
+
+export const messageDigest = message => {
 	const msgBytes = Buffer.from(message, 'utf8');
+	const msgLenBytes = Buffer.from(String(message.length), 'utf8');
```

Comment:
This should be a set number of bytes or delimited in some way.

Reply:
Not necessary, but other cryptos use this way. And instead of `String(message.length)` they use `varint`. Bitcoin use something similar to this https://github.com/LiskHQ/lisk-elements/issues/673#issuecomment-391617664

Reply:
Do you know why they use `varint`? It seems like it would make it possible for people to construct messages that are different but achieve the same `varint(length) + message`.

Reply:
>A VarInt (variable integer) is a field used in transaction data to indicate the
>number of upcoming fields, or the length of an upcoming field.

http://learnmeabitcoin.com/glossary/varint

Reply:
@willclarktech should I update this code with `sha256` and `varint` for message signing, or I need to wait until Iker make research?

Reply:
@alepop As mentioned [here](https://github.com/LiskHQ/lisk-elements/issues/673#issuecomment-391671006) Iker has a few issues to look into before we can move ahead with a solution.